### ISSUE TITLE
[Assistive tech and device support] Pages require sensory characteristics to be understood or operable. (11.11.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#34160]
#### VFS team name
Debt Resolution

#### VFS product name
Medical Copays (MCP)

#### Point of Contact/Reviewers
Josh Kim (@joshkimux) - Accessibility

---

### Platform Issue
Pages require sensory characteristics to be understood or operable.

### Issue Details
No alternative is provided to screen reader users to inform them that content is important (which is demonstrated to sighted users through the "!" icon).

### VA.gov Experience Standard
[Category Number 11, Issue Number 11](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 1.3.3_A

---

### Platform Recommendation
Following VA design system guidance, we recommend including a `sr-only` span to provide an alternative to this icon font. For example: `<span class="sr-only">Alert: [or whatever equivalent alternative needed e.g. "Important"]</span>`. For more details, check out this ticket that resolved a similar issue on alt icon text.

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket.

**Comment by LinnJS** [comment]
[here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/medical-copays/components/Alerts.jsx#L31) is the file that we have all Alerts in for Medical Copays. 

One thing to note is we are not using a single icon element in this file.

When looking at the Design System Alert [here](https://github.com/department-of-veterans-affairs/component-library/blob/master/packages/web-components/src/components/va-alert/va-alert.tsx#L150-L163).

So it seems this is most likely a problems across all of our Alert components across VA.gov.

Should probably be a ticket to address with the Design System team.

**Comment by shiragoodman** [comment]
@joshkimux can you look into this when you're back in the office? @LinnJS if confirmed, we will get this ticket transferred appropriately. Thanks!

**Comment by joshkimux** [comment]
Still in the office 😄 

@LinnJS , thanks for exploring this! For some context, I thought this would be fixed in the design system [according to this ticket](https://github.com/department-of-veterans-affairs/component-library/pull/74) and [noted in this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19674#issuecomment-830233729).

Would love @bkjohnson or @caw310 to help out here and determine if this might need a second look 🙏

**Comment by LinnJS** [comment]
Sorry @shiragoodman didn't see the line that said "point of contact".

I am pretty familiar with this accessibility pattern from using TailwindUI Components on my personal projects. If this was a one case kind of senerio I would be fine with opening a PR on the Design System but feels like there could be some edge cases that I dont know about. Having issues running web components locally.

What I would recommend is some kind of switch statement that references all of the Alert "statuses" and returns an `sr-only` span that corresponds to the status of the Alert.

**Comment by bkjohnson** [comment]
@joshkimux the original implementation did have `<span class="sr-only">Alert: </span>`, but that was changed in https://github.com/department-of-veterans-affairs/component-library/pull/137. Based on my understanding of that PR, an `aria-live` attribute was added which made the `sr-only` text become redundant as a way of announcing a change.

**Comment by craigwheeler** [comment]
hey @joshkimux just wanted to touch base with you to find out if this was cleared up or do we still need changes?

**Comment by joshkimux** [comment]
Apologies to folks, still on PTO but I have some downtime at the airport to reply.

@LinnJS and @bkjohnson , I made a mistake in my comment above which turned the conversation towards the va alert component! 🙇 

To clarify, in the original ticket I was referring to the "!" icon in the cards themselves which I don't (think) are VA alert components, hence the missing `sr-only` to announce the alternative text.

I've attached a video of screen reader behavior where the "!" icon is skipped due to it having a `role="presentation"` and no alternative text.

https://user-images.githubusercontent.com/14154792/149221226-41d8281d-b797-45fc-b824-014de639b1db.mov

Hope that clears things up! 🙏

**Comment by bkjohnson** [comment]
@joshkimux looks like you're right, what we're seeing here [isn't a design system component](https://github.com/department-of-veterans-affairs/vets-website/blob/ab0eb9bef817727f7b8f0146f9cae5a3dd2417fc/src/applications/medical-copays/components/BalanceCard.jsx#L27-L43).

Here's an example of what it could like if the alert component from the design system were used:

![image](https://user-images.githubusercontent.com/2008881/149239282-bbfcb39e-0f32-406c-b727-215a73ea399c.png)

Since this is a `warning` alert the `aria-live="polite"` attribute wouldn't be added, but we could discuss changing the a11y behavior if it would be an improvement.

**Comment by denisecoveyduc** [comment]
@joshkimux @shiragoodman  Just following up, sounds like this is a platform issue not a design system component. Do I have the right? If so any idea who we can/should reach out to for this one?

**Comment by shiragoodman** [comment]
@bkjohnson could you elaborate on your latest comment? If it's not a Design System issue, then do you know what Platform team would be responsible? I'd like to make sure we can get this ticket reassigned appropriately.

**Comment by bkjohnson** [comment]
It looked like the team is building their own card component with alert-ish features rather than using the Design System's alert component. I was pointing out that with our component, a `warning` alert doesn't add an `aria-live` attribute that could alert screen readers, but if the team wanted to take advantage of our component we could look into adding that functionality, especially if it would improve a11y.

If Design System components aren't being used, I would categorize this as an application issue rather than anything at the platform level. That `BalanceCard` is a component built in app code that just happens to use an icon provided at the platform level. I'm just basing this off of Josh's video showing that one app component - it's unclear to me if this issue is about components/pages outside of that example.
